src/services/socket/api.ts

```diff
@@ -93,20 +96,68 @@ export class SocketAPI {
     this.isHealthy = null
   }
 
+  /**
+   * Processes Socket API responses and throws custom errors for various
+   * failures, including handling the API's unique response structure.
+   */
+  async #handleResponse<T>({
+    fetchPromise,
+    errorPrefix
+  }: {
+    fetchPromise: Promise<CustomResponse>
+    errorPrefix: string
+  }): Promise<T> {
+    let response: CustomResponse
+
+    try {
+      response = await fetchPromise
+    } catch (e: any) {
+      const message = e?.message || 'no message'
+      const status = e?.status || 'unknown'
+      const error = `${errorPrefix} Upstream error: <${message}>, status: <${status}>`
```

Comment:
Perhaps we can hide Upstream error: `no message` and status: `unknown`.

Reply:
I changed it to displaying the status only if one exists in https://github.com/AmbireTech/ambire-common/pull/1157/commits/f251912586546a977c52002fa8451efea15db7a2, but left the message as is, since I haven't yet had a situation with no message (and it might be actually useful if we see no message from the upstream provider while getting bug reports).